themes/default/content/blog/pulumi-recommended-patterns-the-basics/index.md

```diff
@@ -0,0 +1,177 @@
+
+Of course, while the programming language used to describe your infrastructure desired state is important, so is your integrated development environment (IDE). You're going to spend most of your time in that application, so choose the one you are the most efficient with.
+
+According to the [Stack Overflow survey](https://insights.stackoverflow.com/survey/2021#section-most-popular-technologies-integrated-development-environment) in 2021, the top 5 IDEs were:
+
+1. Visual Studio Code (Free, Open-source)
+2. Visual Studio (Free and paid)
+3. Notepad++ (Free, open-source)
+4. IntelliJ (Free and paid)
+5. Vim (Free, open-source)
+
+If you're unsure which IDE you should be using, here is a list of features you want to consider:
+
+* Code completion for your preferred language
+* Code syntax check and linting
+* Integrated code documentation support
+* Integration with your source code control workflows (git, Mercurial, SVN...)
+* Plugins and extension support to augment the IDE capabilities (issues and pull request management, prettier code, themes and colors, pair programming...)
+
+And just as with the cloud vendors, you should definitely consider the do's, don'ts, and recommended patterns for the programming language you selected. Following these best practices gives you safe guardrails and streamlines your developer experience across teams sharing the same programming language. Here's the main ones for a few languages:
+
+* [TypeScript](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html)
+* Python: [PEP 8](https://www.python.org/dev/peps/pep-0008/), [PEP 20](https://www.python.org/dev/peps/pep-0020/), and [an additional blog post](https://data-flair.training/blogs/python-best-practices/)
+* [C#](https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/coding-conventions)
+* Golang: [Golang docs](https://golangdocs.com/golang-best-practices) and [a talk](https://talks.golang.org/2013/bestpractices.slide#1)
+
+## Pulumi recommended patterns
+
+Now we're all set, so let's dive into writing a Pulumi application.
+
+### Give your resources a name
+
+The first thing you'll notice when you write a Pulumi app is that every single resource must have its own name. This *Pulumi resource name* or a logical resource name if you prefer, is used inside your stack state file to uniquely identify your resources. This is also what you will see at first when exploring your stack resources in the Pulumi web console.
+
+Depending on the resource type, a second name (a *cloud resource name* or a *physical resource name* if you like) may also be provided. This is the name that will be visible in your Cloud vendor web console. If this *cloud resource name* is unspecified, Pulumi uses the *Pulumi resource name* and appends a [random suffix]({{< relref "/docs/intro/concepts/resources/#autonaming" >}}) to it to make it unique.
```

Comment:
Adjustment to remove inaccessible wording. Also, one thing Hugo seems to require with `relref` for future reference: If you want to use an anchor link on another page, you have to remove the trailing slash. I forget exactly why, but something to file away for any future articles :)
```suggestion
Depending on the resource type, a second name (a *cloud resource name* or a *physical resource name* if you like) may also be provided. This name will be what is used in your cloud vendor's web console. If this *cloud resource name* is unspecified, Pulumi uses the *Pulumi resource name* and appends a [random suffix]({{< relref "/docs/intro/concepts/resources#autonaming" >}}) to it to make it unique.
```